tests/UnitTests/Benchmark/BenchmarkTest.h

```diff
@@ -0,0 +1,156 @@
+﻿//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#include <TestFramework.h>
+#include <type_traits>
+#include <chrono>
+#include <numeric>
+#include <utility>
+#include <memory>
+
+namespace testing {
+class BenchmarkTestBase {
+public:
+    inline void Run();
+
+    // Number of times to run each test, default to 1
+    size_t GetRunCount() const {
+        return 1;
+    }
+
+    // Pre-run and Post-run equivalent to SetUp and TearDown per test case
+    // Setup and Tear down that is performed once per all tests should be done in the constructor/destructor
+    void PreRun() {
+        // Default do nothing
+    }
+
+    void PostRun() {
+        // Default do nothing
+    }
+};
+
+namespace internal {
+
+// Helper method shared between BenchmarkTestRunner classes
+// This is called to actually run and benchmark the tests
+// Can't be shared internally because parameterized tests need a different inheritance than non
+template <typename T>
+static void __RunTest(T& test) {
+    using Microseconds = double;
+    size_t runCount = test.GetRunCount();
+    ASSERT_NE(0, runCount);
+    std::vector<Microseconds> results(runCount);
+    for (size_t i = 0; i < runCount; ++i) {
+        test.PreRun();
+        auto start = std::chrono::high_resolution_clock::now();
+        test.Run();
+        auto end = std::chrono::high_resolution_clock::now();
+        test.PostRun();
+        std::chrono::duration<Microseconds, std::micro> duration = end - start;
+        results[i] = duration.count();
+    }
+
+    if (runCount > 1) {
+        // Algorithm for calculating sample mean and variance with minimal rounding errors
+        Microseconds netRuntime = 0.0;
+        Microseconds mean = 0.0;
+        Microseconds q = 0.0;
+        for (size_t i = 0; i < runCount; ++i) {
+            Microseconds previousMean = mean;
+            netRuntime += results[i];
+            mean += (results[i] - mean) / (i + 1);
+            q += (results[i] - previousMean) * (results[i] - mean);
+        }
+
+        Microseconds sampleStandardDeviation = std::sqrt(q / (runCount - 1));
+
+        LOG_INFO(
+            "\nNet Runtime: %f microseconds\nSample Mean Runtime: %f microseconds\nSample Standard Deviation: %f microseconds\n%u test "
```

Comment:
nit: probably out of scope, but I wonder if we would benefit from having a option to output in a csv format... probably overengineering if it looks like we'll eventually change the framework though.